app/src/components/UnitForm/index.js

```diff
@@ -0,0 +1,176 @@
+    setPostData({
+      ...postData,
+      [e.target.name]: e.target.value,
+    });
+  };
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const res = await axios.post("/list_unit", {
+      ...postData,
+    });
+    console.log(res);
+    setPostData(defaultUnitValues);
+
+    handleClose();
+  };
+
+  return (
+    <Paper className={classes.paper}>
+      <form
+        autoComplete="off"
+        noValidate
+        className={classes.form}
+        onSubmit={handleSubmit}
+      >
+        <Typography variant="h5">Unit Information</Typography>
+        <IconButton edge="end" aria-label="Edit" onClick={handleClose}>
+          <CloseIcon />
+        </IconButton>
+        <TextField
+          name="address"
+          variant="outlined"
+          label="Address"
+          fullWidth
+          required
+          value={postData.address}
+          onChange={handleChange}
+        />
+        <TextField
+          name="room"
+          variant="outlined"
+          label="Room"
+          type="number"
+          fullWidth
+          value={postData.room}
+          className={classes.numbers}
+          onChange={handleChange}
+        />
+        <TextField
+          name="price"
+          variant="outlined"
+          label="Price"
+          type="number"
+          className={classes.numbers}
+          required
+          value={postData.price}
+          onChange={handleNum}
+        />
+        <TextField
+          name="numBeds"
+          variant="outlined"
+          label="Bedrooms"
+          type="number"
+          className={classes.numbers}
+          required
+          value={postData.numBeds}
+          onChange={handleNum}
+        />
+        <TextField
+          name="numWashrooms"
+          variant="outlined"
+          label="Washrooms"
+          type="number"
+          className={classes.numbers}
+          required
+          value={postData.numWashrooms}
+          onChange={handleNum}
+        />
+        <TextField
+          name="floor"
+          variant="outlined"
+          label="Floor"
+          type="number"
+          className={classes.numbers}
+          value={postData.floor}
+          onChange={handleNum}
+        />
+        <TextField
+          name="leaseDuration"
+          variant="outlined"
+          label="Lease Duration"
+          fullWidth
+          select
+          required
+          value={postData.leaseDuration}
+          onChange={handleChange}
+        >
+          {leaseOptions.map((option) => (
+            <MenuItem key={option.value} value={option.value}>
+              {option.label}
+            </MenuItem>
+          ))}
+        </TextField>
+        <div>
+          <FileBase
+            type="file"
+            multiple={false}
+            onDone={({ base64 }) =>
+              setPostData({ ...postData, selectedFile: base64 })
+            }
+          />
+        </div>
+        <Button
+          // disabled={
```

Comment:
Leaving a comment here, MUI makes it really hard to prevent a user from submitting a form even if the required fields aren't filled in, so if we wanted to have field validation, we could add a disabled check on the button until all the required fields are entered (the code commented out here).